recipes/approvaltests.cpp/8.0.0/conanfile.py

```diff
@@ -0,0 +1,42 @@
+import os
+from conans import ConanFile, tools
+
+
+class ApprovalTestsCppConan(ConanFile):
+    name = "approvaltests.cpp"
+    version = "8.0.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/approvals/ApprovalTests.cpp"
+    license = "Apache-2.0"
+    description = "Approval Tests allow you to verify a chunk of output " \
+                  "(such as a file) in one operation as opposed to writing " \
+                  "test assertions for each element."
+    topics = ("conan", "testing", "unit-testing", "header-only")
+    options = {"test_framework": ["catch2", "gtest", "doctest"]}
+    default_options = {"test_framework": "catch2"}
+    no_copy_source = True
+
+    @property
+    def _header_file(self):
+        return "ApprovalTests.hpp"
+
+    def requirements(self):
+        if self.options.test_framework == "catch2":
+            self.requires("catch2/2.11.0")
+        elif self.options.test_framework == "gtest":
+            self.requires("gtest/1.10.0")
+        elif self.options.test_framework == "doctest":
+            self.requires("doctest/2.3.5")
+
+    def source(self):
+        for source in self.conan_data["sources"][self.version]:
+            url = source["url"]
+            filename = url[url.rfind("/") + 1:]
+            tools.download(url, filename)
+            tools.check_sha256(filename, source["sha256"])
```

Comment:
Why not `tools.get(**source)` in one single line? Python will convert the dictionary (url, sha256) into named arguments and it will use the same filename

Reply:
`tools.get` works for compressed files only, the download will get a uncompressed file (a header filer) only. So only way is using `tools.download` instead.

Anyway, I don't like this approach, I would like to execute something similar to `tools.get`. We could update `tools.get` accepting uncompressed files too, or add a new argument in `tools.download` to check the sha256

Reply:
Looks like this needs to remain two lines for now. Correct?

Reply:
Yes, uilian was right about the non-zipped files